client/view.js

```diff
@@ -73,18 +68,30 @@ export default class View
 
 	#addPlayer(hand)
 	{
+		const $container = document
 		var $player = this.$playerRowTemplate.cloneNode(true);
 		$player.querySelector('.js-id').textContent = hand.playerId;
 		this.#playerRows[hand.playerId] = $player;
 		$player.querySelector('.js-name').textContent = hand.playerName;
-		$player.querySelector('.js-owner').textContent = hand.gameOwner;
+		$player.querySelector('.js-owner').innerHTML = hand.gameOwnerButton;
 		this.$playerTableBody.appendChild($player);
+		$container.querySelector('.js-deal').addEventListener('click', e => {
+			e.preventDefault();
+	
+			if(hand.gameOwner === true)
+			{
+				this.$txtNumJokers = $player.querySelector('.num-jokers')
+				const numJokers = parseInt(this.$txtNumJokers.value, 10);
+				this.#bus.publish('deal', numJokers);
+				$player.querySelector('.js-owner').textContent = ""

```

Comment:
this button click should just issue a command (deal), it shouldnt be updating the UI as well. The side effect (eg events that come back on the bus) should update the UI

Reply:
I've amended this, it feels ok but also feels like I've littered the code with gameStarted.

Reply:
not sure what you mean, probably worth a chat. 
my point is that interactions, e,g. button clicks, want to Send Commands and nothing else. reason being that you want all game-state and UI side effects to be driven by the consequential Events. If you also put UI side effecting code in the origin of commands (button clicks) then you will also need it in the event handlers (duplication).
Consider that anything could trigger an Event, not just be the result of a Command issued from an interaction (click), e.g. Time might elapse, something another player does, a change in the weather, anything can trigger an event, so only update the UI based off events then you have a single place to do it from and its nice and tidy and contained. If you _only_ do it from a button click then events cant trigger it - ok it may be the case right now for a given thing that this is ok, but its a bad way to work as your basically writing future bugs right now

Reply:
Unless I've completely missed the point, I think what I've done satisfies this. 

- The button click initiates the 'deal' command and publishes this to the bus
- Client controller receives this from the bus and sends a 'deal' command
- Server controller receives this command and gets the players' hands from game.js
- In game.js the gameStarted property flicks to true (the game owners hand owns this property)
- This flows back to the server controller and runs updateAllPlayersHands
- I get a bit hazy with websockets here, but this gets fed back to view.js and updates the game owners hand and hides the deal button based on the gameStarted property. 

Might be worth a chat if I'm not making sense.

Reply:
thats it, so delete this line:
`$player.querySelector('.js-owner').textContent = ""`
because youre updating the UI from the button click, which isnt what we describe above, this should happen in the last step
(note there are "exceptions" to this like showing a "loading" screen as you submit the command and wait for the event to come back, but we'll deal with that later)

Reply:
great, I'd already removed this a couple of commits ago (but after the start of this conversation) , this was the commit - https://github.com/Tom91011/big-2-game/pull/18/commits/8fbb7bdc7c844f0cc0926f38f433a1e780c71d04